js/renovation/ui/overlays/popover.tsx

```diff
@@ -0,0 +1,158 @@
+
+export const viewFunction = ({
+  wrapperRef,
+  props,
+  restAttributes,
+}: Popover): JSX.Element => {
+  const { children } = props;
+
+  return (
+    <DomComponentWrapper
+      componentType={LegacyPopover}
+      componentProps={props}
+      ref={wrapperRef}
+      // eslint-disable-next-line react/jsx-props-no-spreading
+      {...restAttributes}
+    >
+      {children}
+    </DomComponentWrapper>
+  );
+};
+
+@ComponentBindings()
+export class PopoverProps extends BaseWidgetProps {
+  @OneWay() animation?: {
+    show: AnimationConfig;
+    hide: AnimationConfig;
+  } = {
+    show: { type: 'fade', from: 0, to: 1 },
+    hide: { type: 'fade', to: 0 },
+  };
+
+  @OneWay() closeOnOutsideClick: boolean | (() => boolean) = false;
+
+  @OneWay() container?: string | Element;
+
+  // contentComponent
+
+  // contentRender
+
+  @Template() contentTemplate: template | ((contentElement: DxElement) => string | UserDefinedElement) = 'content';
+
+  @OneWay() deferRendering = true;
+
+  @OneWay() disabled = false;
+
+  @OneWay() height: number | string | (() => number | string) = 'auto';
+
+  @OneWay() hideEvent?: string;
+
+  @OneWay() hint?: string;
+
+  @OneWay() hoverStateEnabled = false;
+
+  @OneWay() maxHeight?: number | string | (() => number | string) | null = null;
+
+  @OneWay() maxWidth?: number | string | (() => number | string) | null = null;
+
+  @OneWay() minHeight?: number | string | (() => number | string) | null = null;
+
+  @OneWay() minWidth?: number | string | (() => number | string) | null = null;
+
+  @Event() onHidden?: () => void;
+
+  @Event() onHiding?: () => void;
+
+  @Event() onInitialized?: () => void;
+
+  @Event() onOptionChanged?: () => void;
+
+  @Event() onShowing?: () => void;
+
+  @Event() onShown?: () => void;
+
+  @Event() onTitleRendered?: () => void;
+
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  @OneWay() position: PositionConfig | Function
+  | 'bottom' | 'left' | 'right' | 'top' = 'bottom';
+
+  @OneWay() rtlEnabled = false;
+
+  @OneWay() shading = false;
+
+  @OneWay() shadingColor = '';
+
+  @OneWay() showCloseButton = isDesktop;
+
+  @OneWay() showEvent?: string;
+
+  @OneWay() showTitle = false;
+
+  @OneWay() target?: string | Element; // TODO: default value
+
+  @OneWay() title = '';
+
+  // titleComponent
+
+  // titleRender
+
+  @Template() titleTemplate: template | ((contentElement: DxElement) => string | UserDefinedElement) = 'title';
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  @OneWay() toolbarItems?: any;
+
+  @TwoWay() visible = true;
+
+  @OneWay() width: number | string | (() => number | string) = 'auto';
+
+  @Slot() children?: JSX.Element;
+}
+@Component({
+  view: viewFunction,
+})
+export class Popover extends JSXComponent<PopoverProps, 'visible'>() {
+  @Ref() wrapperRef!: RefObject<DomComponentWrapper>;
+
+  @Mutable() // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
mb this way?

```suggestion
  @Mutable()  instance!: { option }; 
```

and `this.instance = this.wrapperRef.current?.getInstance() as { option };` below.

no eslint-disable is need if we write the code this way